lib/hotel.rb

```diff
@@ -0,0 +1,163 @@
+require_relative 'room'
+require_relative 'reservation'
+
+module HotelSystem 
+  class Hotel 
+    attr_accessor :rooms
+
+    def initialize
+      @rooms = [] 
+      20.times do |i|
+        @rooms << HotelSystem::Room.new(i+1, 200)
+      end 
+    end 
+    #the block can only be the last 5 rooms 15-20
+    # def make_reservation_block(start_date, end_date, rooms_numbers, rate)
+    #   rooms = []
+    #   rooms_numbers.each do |num|
+    #     if num > 20 || num < 15
+    #       raise ArgumentError.new("you can only reserve rooms 15-20 for a block reservation")
+    #     end 
+    #     rooms << find_room(num)
+    #   end 
+      
+    #   rooms.each do |room|
+    #     room.cost_per_night = cost_with_discount(rate, room)
+    #     make_reservation(room.room_number, start_date, end_date)
+    #   end
+    # end 
+    
+    def find_reservations_with_room_number(room_number, start_date, end_date)
+      room = find_room(room_number)
+      list_reservations = []
+        room.reservations.each do |reservation|
+          if is_between_two_dates([start_date, end_date], reservation.start_date) || is_between_two_dates([start_date, end_date], reservation.end_date) 
+            list_reservations << reservation
+          end
+        end 
+      return list_reservations
+    end 
+
+    def list_reservations_dates(start_date, end_date) 
+      list_reservations = []
+      @rooms.each do |room|
+        room.reservations.each do |reservation|
+          if is_between_two_dates([start_date, end_date], reservation.start_date) || is_between_two_dates([start_date, end_date], reservation.end_date) 
+            list_reservations << reservation
+          end 
+        end 
+      end  
+      return list_reservations
+    end 
+    
+    def find_room(room_number)
+      @rooms.each do |room| 
+        if room_number == room.room_number 
+          return room
+        end 
+      end
+    end
+
+    def find_reservation(reservation_number)
+      @rooms.each do |room|
+        room.reservations.each do |reservation|
+          if reservation.reservation_number == reservation_number
+            return reservation
+          end 
+        end 
+      end
```

Comment:
If you're going to have `reservations` stored in `Room` you should have this functionality inside of a method there instead of here:

```suggestion
      @rooms.each do |room|
        room.find_reservation(reservation_number)
      end
```

That properly encapsulates this functionality and means that `Hotel` has to know less about `Room`.